docker/etc/nginx/apps/drupal/drupal.conf

```
@@ -21,22 +21,29 @@ location / {
 
     ## Using a nested location is the 'correct' way to use regexes.
 
-    ## Regular private file serving (i.e. handled by Drupal).
-    location ^~ /(files|media)/ {
-        ## Include the specific FastCGI configuration. This is for a
-        ## FCGI backend like php-cgi or php-fpm.
-        include apps/drupal/fastcgi_drupal.conf;
-        fastcgi_pass phpcgi;
+    ## Handle requests to get a file's content.
+    location ~ "^/files/([^/]{2})([^/]{2})([^/]{32})/([^/]+)(\.[^./]+)$" {
```

This ensures the location *only* matches a valid UUID, not any random 36 character string. The ~* makes the regex case insensitive so it can be a bit shorter and does not need to list `a-f` as well.  I changed the `filename.ext` part, so we can handle files with full stops in their name. Maybe we should not, because Windows, but that is a discussion for later :-)
```suggestion
    location ~* ^/files/([0-9A-F]{2})([0-9A-F]{2})([0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12})/([^\.]+)\.([^\.]+)$
```